ResearchKit/Common/ORKHeadlineLabel.m

```diff
@@ -53,6 +55,17 @@ + (UIFont *)defaultFont {
     return [self defaultFontInSurveyMode:NO];
 }
 
+- (instancetype)initWithFrame:(CGRect)frame {
+    [NSNotificationCenter.defaultCenter addObserverForName:CEVORKStepViewControllerViewWillAppearNotification object:nil queue:NSOperationQueue.mainQueue usingBlock:^(NSNotification * _Nonnull note) {
+        CEVRKTheme *theme = note.userInfo[CEVRKThemeKey];
+        if ([theme isKindOfClass:[CEVRKTheme class]]) {
+            self.cev_theme = theme;
```

Comment:
Strong `self` reference here will retain the ORKHeadlineLabel after its view controller is gone. Would be safer to use a weak-self pattern within the notification block

Reply:
good catch!